libbeat/idxmgmt/idxmgmt.go

```diff
@@ -99,37 +99,47 @@ func DefaultSupport(log *logp.Logger, info beat.Info, configRoot *config.C) (Sup
 
 // MakeDefaultSupport creates some default index management support, with a
 // custom ILM support implementation.
-func MakeDefaultSupport(ilmSupport ilm.SupportFactory) SupportFactory {
+func MakeDefaultSupport(ilmSupport lifecycle.SupportFactory) SupportFactory {
 	if ilmSupport == nil {
-		ilmSupport = ilm.DefaultSupport
+		ilmSupport = lifecycle.DefaultSupport
 	}
 
 	return func(log *logp.Logger, info beat.Info, configRoot *config.C) (Supporter, error) {
 		const logName = "index-management"
+		if log == nil {
+			log = logp.NewLogger(logName)
+		} else {
+			log = log.Named(logName)
+		}
 
+		// now that we have the "correct" default, unpack the rest of the config
 		cfg := struct {
-			ILM       *config.C        `config:"setup.ilm"`
-			Template  *config.C        `config:"setup.template"`
-			Output    config.Namespace `config:"output"`
-			Migration *config.C        `config:"migration.6_to_7"`
+			Lifecycle lifecycle.RawConfig `config:",inline"`
+			Template  *config.C           `config:"setup.template"`
+			Output    config.Namespace    `config:"output"`
+			Migration *config.C           `config:"migration.6_to_7"`
 		}{}
 		if configRoot != nil {
 			if err := configRoot.Unpack(&cfg); err != nil {
-				return nil, err
+				return nil, fmt.Errorf("error unpacking cfg settings while setting up index support: %w", err)
 			}
 		}
 
-		if log == nil {
-			log = logp.NewLogger(logName)
-		} else {
-			log = log.Named(logName)
+		// consider lifecycles enabled if the user has explicitly enabled them,
+		// or if no `enabled`` setting has been set by the user, thus reverting to a default of enabled.
+		enabled := false
+		if cfg.Lifecycle.DSL.Enabled() || cfg.Lifecycle.ILM.Enabled() {
+			enabled = true
+		}
+		if (cfg.Lifecycle.DSL == nil || !cfg.Lifecycle.DSL.HasField("enabled")) && (cfg.Lifecycle.ILM == nil || !cfg.Lifecycle.ILM.HasField("enabled")) {
+			enabled = true
```

Comment:
So, the weird `nil` checks are just because the `config` library is weird. `Enabled()` checks to see if the underlying object is nil before accessing it, but `HasField()` does not, hence the extra checks.

Reply:
There should be unit tests for this--if you remove the `== nil` checks, the unit tests should panic.